lib/Models/REpresentationalStateTransferCatalogFunction.js

```diff
@@ -0,0 +1,540 @@
+                crs: usedCrs
+            });
+        },
+        functionParameterToInput: function(catalogFunction, parameter, value) {
+            var bboxMinCoord1, bboxMinCoord2, bboxMaxCoord1, bboxMaxCoord2, urn;
+            // We only support CRS84 and EPSG:4326
+            if (parameter.crs.indexOf('crs84') !== -1) {
+                // CRS84 uses long, lat rather that lat, long order.
+                bboxMinCoord1 = CesiumMath.toDegrees(value.west);
+                bboxMinCoord2 = CesiumMath.toDegrees(value.south);
+                bboxMaxCoord1 = CesiumMath.toDegrees(value.east);
+                bboxMaxCoord2 = CesiumMath.toDegrees(value.north);
+                // Comfortingly known as WGS 84 longitude-latitude according to Table 3 in OGC 07-092r1.
+                urn = 'urn:ogc:def:crs:OGC:1.3:CRS84';
+
+            } else {
+                // The URN value urn:ogc:def:crs:EPSG:6.6:4326 shall mean the Coordinate Reference System (CRS) with code
+                // 4326 specified in version 6.6 of the EPSG database available at http://www.epsg.org/. That CRS specifies
+                // the axis order as Latitude followed by Longitude.
+                // We don't know about other URN versions, so are going to return 6.6 regardless of what was requested.
+                bboxMinCoord1 = CesiumMath.toDegrees(value.south);
+                bboxMinCoord2 = CesiumMath.toDegrees(value.west);
+                bboxMaxCoord1 = CesiumMath.toDegrees(value.north);
+                bboxMaxCoord2 = CesiumMath.toDegrees(value.east);
+                urn = 'urn:ogc:def:crs:EPSG:6.6:4326';
+            }
+
+            return {
+                       inputType: 'BoundingBoxData',
+                       inputValue: bboxMinCoord1 + ',' + bboxMinCoord2 + ',' + bboxMaxCoord1 + ',' + bboxMaxCoord2 + ',' + urn
+                   };
+        }
+    }
+   ];
+
+REpresentationalStateTransferCatalogFunction.prototype._load = function() {
+    var inputs = this.input;
+    if (!Array.isArray(inputs)) {
+        inputs = [inputs];
+    }
+    this._parameters = inputs.map(createParameterFromRestInput.bind(undefined, this));
+};
+
+/**
+ * Invoke the REST function with the provided parameterValues.
+ * @return {Promise}
+ */
+REpresentationalStateTransferCatalogFunction.prototype.invoke = function() {
+    var now = new Date();
+    var timestamp = sprintf('%04d-%02d-%02dT%02d:%02d:%02d', now.getFullYear(), now.getMonth() + 1, now.getDate(), now.getHours(), now.getMinutes(), now.getSeconds());
+
+    var asyncResult = new ResultPendingCatalogItem(this.terria);
+    asyncResult.name = this.name + ' ' + timestamp;
+    asyncResult.description = 'This is the result of invoking the ' + this.name + ' process or service at ' + timestamp + ' with the input parameters below.';
+    var inputsSection =
+        '<table class="cesium-infoBox-defaultTable">' +
+        (this.parameters || []).reduce(function(previousValue, parameter) {
```

Comment:
This is the only place `this.parameters` is used.  Should it be `this._parameters`?